src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementEventService.java

```diff
@@ -95,39 +97,67 @@ public EventSummary createEvent(final Integer projectId, final CreateEvent creat
+            project.getLotNumber(), createEventNonOCDS.getEventType(), principal);
+        eventBuilder.assessmentId(newAssessmentId);
+        returnAssessmentId = newAssessmentId;
+        log.debug("Created new empty assessment: {}", newAssessmentId);
+      } else {
+        var validatedAssessment =
+            assessmentService.getAssessment(createEvent.getNonOCDS().getAssessmentId(), principal);
+        eventBuilder.assessmentId(validatedAssessment.getAssessmentId());
+        returnAssessmentId = validatedAssessment.getAssessmentId();
+        log.debug("Linking existing assessment: {} to new event",
+            validatedAssessment.getAssessmentId());
+      }
+    } else {
+
+      var createUpdateRfx = createRfxRequest(project, eventName, principal);
+
+      var createRfxResponse =
+          ofNullable(jaggaerWebClient.post().uri(jaggaerAPIConfig.getCreateRfx().get(ENDPOINT))
+              .bodyValue(createUpdateRfx).retrieve().bodyToMono(CreateUpdateRfxResponse.class)
+              .block(ofSeconds(jaggaerAPIConfig.getTimeoutDuration())))
+                  .orElseThrow(() -> new JaggaerApplicationException(INTERNAL_SERVER_ERROR.value(),
+                      "Unexpected error creating Rfx"));
 
-    if (createRfxResponse.getReturnCode() != 0
-        || !Constants.JAGGAER_GET_OK_MSG.equals(createRfxResponse.getReturnMessage())) {
-      log.error(createRfxResponse.toString());
-      throw new JaggaerApplicationException(createRfxResponse.getReturnCode(),
-          createRfxResponse.getReturnMessage());
+      if (createRfxResponse.getReturnCode() != 0
+          || !Constants.JAGGAER_GET_OK_MSG.equals(createRfxResponse.getReturnMessage())) {
+        log.error(createRfxResponse.toString());
+        throw new JaggaerApplicationException(createRfxResponse.getReturnCode(),
+            createRfxResponse.getReturnMessage());
+      }
+      log.info("Created Jaggaer (Rfx) event: {}", createRfxResponse);
+      rfxReferenceCode = createRfxResponse.getRfxReferenceCode();
+      eventBuilder.externalEventId(createRfxResponse.getRfxId())
+          .externalReferenceId(createRfxResponse.getRfxReferenceCode());
     }
-    log.info("Created event: {}", createRfxResponse);
 
     // Persist the Jaggaer Rfx details as a new event in the tenders DB
     var ocdsAuthority = ocdsConfig.getAuthority();
     var ocidPrefix = ocdsConfig.getOcidPrefix();
 
-    var event = ProcurementEvent.builder().project(project).eventName(eventName)
-        .externalEventId(createRfxResponse.getRfxId()).eventType(eventTypeValue)
-        .downSelectedSuppliers(downselectedSuppliers)
-        .externalReferenceId(createRfxResponse.getRfxReferenceCode())
-        .ocdsAuthorityName(ocdsAuthority).ocidPrefix(ocidPrefix).createdBy(principal)
-        .createdAt(Instant.now()).updatedBy(principal).updatedAt(Instant.now()).build();
+    var event = eventBuilder.project(project).eventName(eventName).eventType(eventTypeValue)
```

Comment:
Do we not still need the Jaggaer external refs in the case where it is not an assessment event?

Reply:
I think I've got those in on lines 143 and 144 - just above?

Reply:
Ah yep - I see (sorry - was later last night 😄 ) 👍